File: api/v1/schemas/monitor.py
# -*- coding: utf-8 -*-
from pydantic import BaseModel
from typing import List, Optional

class MonitorSummaryItem(BaseModel):
    stock_code: str
    stock_name: str
    operation_advice: str
    trend_prediction: str
    sentiment_label: str
    sentiment_score: int
    key_points: str
    current_price: Optional[float] = None
    change_pct: Optional[float] = None
    update_time: str

class MonitorSummaryResponse(BaseModel):
    total: int
    items: List[MonitorSummaryItem]

class RealtimeMonitorRequest(BaseModel):
    stock_code: str

class RealtimeMonitorResponse(BaseModel):
    stock_code: str
    stock_name: str
    current_price: Optional[float] = None
    change_pct: Optional[float] = None
    advice: str
    update_time: str
    success: bool
    error_message: Optional[str] = None

File: apps/dsa-web/src/api/monitor.ts
import apiClient from './index';

export interface MonitorSummaryItem {
  stock_code: string;
  stock_name: string;
  operation_advice: string;
  trend_prediction: string;
  sentiment_label: string;
  sentiment_score: number;
  key_points: string;
  current_price?: number;
  change_pct?: number;
  update_time: string;
}

export interface MonitorSummaryResponse {
  total: number;
  items: MonitorSummaryItem[];
}

export interface RealtimeMonitorResponse {
  stock_code: string;
  stock_name: string;
  current_price?: number;
  change_pct?: number;
  advice: string;
  update_time: string;
  success: boolean;
  error_message?: string;
}

export const getMonitorSummary = async (): Promise<MonitorSummaryResponse> => {
  const response = await apiClient.get<MonitorSummaryResponse>('/api/v1/monitor/summary');
  return response.data;
};

export const getRealtimeMonitor = async (stockCode: string): Promise<RealtimeMonitorResponse> => {
  const response = await apiClient.post<RealtimeMonitorResponse>('/api/v1/monitor/realtime', {
    stock_code: stockCode,
  });
  return response.data;
};
